conans/client/rest/conan_requester.py

```diff
@@ -58,14 +58,31 @@ def _add_kwargs(self, url, kwargs):
         return kwargs
 
     def get(self, url, **kwargs):
-        return self._requester.get(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("get", url, **kwargs)
 
     def put(self, url, **kwargs):
-        return self._requester.put(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("put", url, **kwargs)
 
     def delete(self, url, **kwargs):
-        return self._requester.delete(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("delete", url, **kwargs)
 
     def post(self, url, **kwargs):
-        return self._requester.post(url, **self._add_kwargs(url, kwargs))
-
+        return self._call_method("post", url, **kwargs)
+
+    def _call_method(self, method, url, **kwargs):
+        old_env = dict(os.environ)
```

Comment:
Put the copy of ``os.environ`` inside the ``if self.proxies``, memorize if something was really popped and check it in ``finally`` to restore them. So we are completely sure not messing with the ``os.environ`` if not strictly necessary.